### DISCUSSION TITLE
Commits

### DISCUSSION THREAD

**Discussion by iLakshya** [community/community#33972] [Pull Requests]
What should I do if I see 1 commit ahead and 1 commit behind of mains branch?
What should I do if I see that your branches have diverged from the main?
What should I do after merging the branch? Should I delete It or should I store it for future use?
Should I close the pull request or merge the branch or let it remain open?

These are certain questions that came to my mind while I was pushing the code. Please help GitHub Community with my doubts.

Happy Learning...✌🏻

**Comment by airtower-luna** [comment]
> What should I do if I see 1 commit ahead and 1 commit behind of mains branch?
> What should I do if I see that your branches have diverged from the main?

There is no general answer here, because it depends on what you want to do with your branches. *If* you want to bring changes from one branch to the other merging is the thing to do, in some cases you might want to rebase instead.

> What should I do after merging the branch? Should I delete It or should I store it for future use?

That's also up to you. Feature branches are often deleted after merging simply because they aren't needed any longer, but if you want to keep them around you can. If you do that a lot you might end up with a confusing amount of old branches, though. :sweat_smile: 

> Should I close the pull request or merge the branch or let it remain open?

This one's simpler: You should merge if you want to add the changes to the target branch as they currently are. If you decide you don't want to merge those changes *ever* (not even after a few more revisions) close the PR. And otherwise (e.g. if you want to work on the changes some more, or are waiting for review) leave it open. Mind that you can mark a PR as a *draft* to let others know that the PR isn't ready for merging yet.

I also recommend you check the "[Git Book](https://git-scm.com/book/en/v2)" site. Chapter 3 is dedicated to working with branches. :slightly_smiling_face:
